src/contribute-to-tidb/contribute-code.md

```diff
@@ -1,88 +1,53 @@
 
-## Looking for what to contribute
+For newcomers, you can check the [starter issues](https://github.com/pingcap/tidb/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22), which are annotated with a "good first issue" label. These are issues suitable for new contributors to work with and won't take long to fix. But because the label is typically added at triage time it can turn out to be inaccurate, so do feel free to leave a comment if you think the classification no longer applies.
 
-If you have a good idea for the contribution, you can proceed to the code contribution process. If you are looking for what you could contribute, you can browse open [GitHub issues](https://github.com/pingcap/tidb/issues) of TiDB, which are not assigned, and then follow the code contribution process. If you are very new to the TiDB project and want to learn about it and its contribution process, you can check the [starter issues](https://github.com/pingcap/tidb/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22), which are annotated with a "good first issue" label.
+To get your change merged you need to sign the [CLA](https://cla-assistant.io/pingcap/tidb?pullRequest=5567) to grant PingCAP ownership of your code.
 
-## Code Contribution Process
+## Contributing process
 
-### TL;DR
+After a consensus is reached in issues, it's time to start the code contributing process:
 
-1. Discuss. Create a GitHub issue or create a topic on the [TiDB Internals forum](https://internals.tidb.io/) and reach consensus.
-2. Implement. Implement the change according to the approach agreed upon in the discussion.
-3. Review. Open a pull request and work with the reviewer.
-4. Merge. A committer of TiDB checks if the contribution fulfills the requirements and merges the code to the codebase.
+  1. Assign the issue to yourself via [\assign](https://prow.tidb.io/command-help?repo=pingcap%2Ftidb#assign). This lets other contributors know you are working on the issue so they won't make duplicate efforts.
+  2. Follow the [GitHub workflow](https://guides.github.com/introduction/flow/), commit code changes in your own git repository branch and open a pull request for code review.
+  3. Make sure the continuous integration checks on your pull request are green (i.e. successful).
+  4. Review and address [comments on your pull request](https://devguide.python.org/pullrequest/). If your pull request becomes outdated with the target branch, you need to [rebase your pull request](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#perform-a-rebase) to keep it up to date.
+  5. When your pull request gets enough approvals(the default number is 2) and all other requirements are met, it will be merged.
+  6. Handle regressions introduced by your change. Although committers bear the main responsibility to fix regressions, it's quite nice for you to handle it(reverting the change or sending fixes).
```

Comment:
Several comments:

1. I don't think we should link to Python Dev Guide here. If we want any of its content, write our own.
2. Because of we use "squash and merge" we don't have to rebase on master but often just merge master to catch up change.

Reply:
1. It's my mistake. Seems I paste a wrong link, it should be a link to a official GitHub guide.
2. Git practise could vary between people. merge and rebase should both work here.